Working server so far

MrMackan opened the pull request.

MrMackan committed:
Trying to connect to DB

Not working completely yet, pushing to allow more testing

MrMackan committed:
Trying to connect to DB

Not working completely yet, pushing to allow more testing

MrMackan committed:
Can now post and get data to and from the DB

Added some test data that can be pushed to the table or create table itself as well. Can also find all data. Will add more features later.
Co-authored-by: erikkellgren1337 <74982427+erikkellgren1337@users.noreply.github.com>

MrMackan committed:
Can now post and get data to and from the DB

Added some test data that can be pushed to the table or create table itself as well. Can also find all data. Will add more features later.
Co-authored-by: erikkellgren1337 <74982427+erikkellgren1337@users.noreply.github.com>

MrMackan committed:
Merge remote-tracking branch 'origin/topic-server' into topic-server

MrMackan committed:
Updated pom.xml to support mqtt

Using Paho to communicate with the mqtt broker.
Co-authored-by: SpaceCrab <10452861+SpaceCrab@users.noreply.github.com>

EricEjfrid committed:
MqttHandler.java added

MqttHandler.java will handle the subscription of mqtt data from the
embedded system

EricEjfrid committed:
Moved MqttHandler.java into controller package

MrMackan committed:
Removed nonworking code

Can start mqtt and spring at same time now

EricEjfrid committed:
Added propertychangesupport in MqttHandler.java

propertychangesupport added to listen for messages and send them to
WeatherController

implementation in weathercontroller pending

erikkellgren1337 committed:
Added comments on DataUI and DataReceiver

Co-authored-by: Mackan <76006738+mrmackan@users.noreply.github.com>

EricEjfrid committed:
Fix dependency issues

added Jaxb-api in pom.xml

MrMackan committed:
Added battery voltage to the DB

EricEjfrid committed:
MqttHandler Database Functionality Added

MqttHandler.java is now implemented as a springApp
MqttHansler can now store data in the database
MqttHandler can now receive multi-line mqtt messages

EricEjfrid committed:
MqttHandler.java Documentation sample.yml Update

Added javadoc comments to MqttHandler.java
Removed dead imports from MqttHandler.java
Updated sample.yml to contain the proper information

EricEjfrid committed:
Added additional instructions to sample.yml

Added instructions in sample.yml on how to configure it

EricEjfrid committed:
Minor fix in sample.yml

<postgres> -> postgres

MrMackan committed:
Updated format, comments and imports

EricEjfrid committed:
MqttHandler.java Now Uses application.yml

Mqtthandler now gets it's connection data from application.yml insttead
of options.yml

portersky commented:
The MqttHandler needs to be reading the application.yml instead of options

EricEjfrid committed:
Merge branch 'topic-server' of github.com:Grohiik/WeatherStation into topic-server

portersky commented:
The paho directories are being created in the WeatherLogger directory and should be inside a data directory instead.

portersky committed:
Merge branch 'main' into topic-server

EricEjfrid committed:
MqttHandler.java now uses data Header

the MqttHandler now sends the data to the database according to the
header of the mqtt-data
paho now stores it's in-flight data in /data

EricEjfrid committed:
Hotfix MqttHandler.java

changed /data to data

portersky committed:
Bring back HelloController.java

portersky reviewed (approved):
Approve for first working code in server.
